modelon/impact/client/entities/workspace.py

```diff
@@ -1126,3 +1210,53 @@ def from_conversion_operation(
     ) -> Workspace:
         assert isinstance(operation, WorkspaceConversionOperation)
         return cls(**kwargs, service=operation._sal)
+
+    @property
+    def received_from(self) -> ReceivedFromWorkspace:
+        """Returns the ReceivedFromWorkspace class."""
+        return ReceivedFromWorkspace(self)
+
+    def get_published_workspace(
+        self, model_name: Optional[str] = None
+    ) -> Optional[PublishedWorkspace]:
+        """Returns the uploaded published workspace.
+
+        Args:
+            model_name: Class path of the modelica model. If specified,
+             an app mode published workspace corresponding to the local workspace
+             is returned(if it exists).
+
+        Returns:
+            An PublishedWorkspace class object.
+
+        Example::
+
+            published_ws = workspace.get_published_workspace()
+
+        """
+        recieved_from = self.definition.received_from
+        if recieved_from:
+            raise RemotePublishedWorkspaceLinkError(
+                "The workspace is linked to a published workspace with ID:"
+                f"{recieved_from.sharing_id}. Please rename the workspace to unlink"
+                "it from the published workspace, if you wish to publish it!"
+            )
+        user = self._sal.users.get_me()["data"]
+        data = self._sal.workspace.get_published_workspaces(
+            name=self.name,
+            owner_username=user["username"],
+            type=PublishedWorkspaceType.APP_MODE.value
+            if model_name
+            else PublishedWorkspaceType.ARCHIVE.value,
+        )["data"]["items"]
+        if not data:
+            logger.warning(
+                "No published workspace corresponding to the local workspace with"
+                f" ID: {self.id} exist!"
+            )
+            return None
+        return PublishedWorkspace(
+            data[0]["id"],
+            definition=PublishedWorkspaceDefinition.from_dict(data[0]),
+            service=self._sal,
+        )
```

Comment:
In the app mode case: can you have several workspaces for different models for the same workspace name? I think you need to check for matching "model"